File: services/auth-service/internal/model/permission.go
package model

// Permission 权限（code = resource:action[:scope]）。与 DB permissions 表对齐。
// 运行时授权只消费 Code；其余字段供管理端点展示与角色权限配置。
type Permission struct {
	ID          int64   `json:"id" db:"id"`
	Code        string  `json:"code" db:"code"`
	Name        string  `json:"name" db:"name"`
	Resource    string  `json:"resource" db:"resource"`
	Action      string  `json:"action" db:"action"`
	Scope       *string `json:"scope" db:"scope"`
	Description string  `json:"description" db:"description"`
}

File: web/src/types/permission.ts
export interface Role {
  id: number;
  code: string;
  name: string;
  description: string;
  parentId: number | null;
  level: number;
  isSystem: boolean;
}

export interface Permission {
  id: number;
  code: string;
  name: string;
  resource: string;
  action: string;
  scope: string | null;
  description: string;
}

export interface CreateRoleRequest {
  code: string;
  name: string;
  description?: string;
  parentCode?: string;
}

export interface CreatePermissionRequest {
  code: string;
  name: string;
  resource: string;
  action: string;
  scope?: string;
  description?: string;
}

export interface RoleDetail {
  role: Role;
  permissions: Permission[];
}
